### PR TITLE
[C++][Restbed] Fix default values for Restbed Server generator

### LINKED ISSUE #760: [C++][Restbed-server] Compilation error from missing function call parameter
##### Description
The following compilation error occurs when I'm trying to compile a Restbed server from the given specification:
```
/api/FrameApi.cpp:70:91: error: expected primary-expression before ‘)’ token
    const bool includePictureBase64 = request->get_query_parameter("includePictureBase64", );
```

##### openapi-generator version
3.2.1(latest master branch with PR #752)

##### OpenAPI declaration file content or url
https://raw.githubusercontent.com/deepviss/deepviss-server-example/master/src/main/resources/deep-viss.json

##### Command line used for generation
Inside cloned repository, after mvn clean package command:
```
java -jar modules/openapi-generator-cli/target/openapi-generator-cli.jar generate/
        -i deep-viss.json /
        -g cpp-restbed-server /
        -o ../openapi-restbed-server
```

##### Steps to reproduce
Assuming Restbed is installed on local machine https://github.com/Corvusoft/restbed
I used the following cmake file:
```
cmake_minimum_required(VERSION 3.10)
project(restbed_ex)

set(CMAKE_CXX_STANDARD 11)

file(GLOB_RECURSE src_files mode/*.cpp api/*.cpp)

add_executable(restbed_ex main.cpp ${src_files})
target_include_directories(restbed_ex PRIVATE /restbed/include/)
target_include_directories(restbed_ex PRIVATE model/)
target_include_directories(restbed_ex PRIVATE api/)
target_link_libraries(restbed_ex restbed)
```
1)Go to output folder
2)Make a dummy main.cpp (touch main.cpp)
3)mkdir build && cd build
4)cmake ..
5)make

### PR DESCRIPTION
**Pull request by stkrwork**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Fix #760

@ravinikam @fvarose @etherealjoy @MartinDelille @wing328

### CODE REVIEW

**Comment by stkrwork** [comment]
Currently what seems to be not working is that if the variable is a string, the default value specified in the java  file is not taken, but if it is a bool for example, it does work.

**Comment by etherealjoy** [comment]
@stkrwork 
I tried debugging the case for the `string` and strangely, `toDefaultValue` is called for `string` but the `{{{DefaultValue}}}` is having nothing.

**Comment by stkrwork** [comment]
So somewhere in the codegen base it doesnt get forwarded?

**Comment by wing328** [comment]
I just fixed the default value in C# (12.3F instead of 12.3). I can take a look at this tomorrow.

**Comment by wing328** [comment]
UPDATE: I've pushed the fix (80b3eff). Please take a look.

Since the fix is in DefaultCodegen, I've copied the core team as well.

cc @OpenAPITools/generator-core-team

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppRestbedServerCodegen.java [hunk 1]
```diff
@@ -287,25 +291,63 @@ public String getTypeDeclaration(Schema p) {
     @Override
     public String toDefaultValue(Schema p) {
         if (ModelUtils.isStringSchema(p)) {
-            return "\"\"";
+            if (p.getDefault() != null) {
```
[the remark this answers is not in the document]
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppRestbedServerCodegen.java]: I think most other generators are following this approach but I'll double check to make sure.

I'm thinking about another PR to clean up the default, example value as the fix in DefaultCodegen in this PR seems to be partial (which means other generators are showing different default, example values after the change)

**Comment by stkrwork** [comment]
looks good to me

**Comment by stkrwork** [comment]
can you check if your problem is fixed  @vbs96?

**Comment by wing328** [comment]
@stkrwork as discussed, we'll need to revert this for the time being as other generators (e.g. Java) are impacted by this.

**Pull request merged**